### PR TITLE
When searching, build the path objects asynchronously while returning the results

### LINKED ISSUE #7264: Speed up search with CouchDB
**Is your feature request related to a problem? Please describe.**
Search is taking a long time to return with CouchDB.

**Describe the solution you'd like**
A quicker search, possibly making the search algorithm more asynchronous.

### PR DESCRIPTION
**Pull request by scottbell**
Closes #7264 

### Describe your changes:

- When building the search results, return them as soon as we can, and asynchronously build the paths incrementally as we receive them.
- Asynchronously search for tags
- Pass abort controllers to search for more details about objects. For ObjectPath, if we've been dismounted, abort the original path search.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by scottbell** [comment]
Before:

https://github.com/nasa/openmct/assets/9853862/642c7b5a-f822-49cb-883e-07f5b6850470

Note all the results, object paths, and annotations come at once.

After:

https://github.com/nasa/openmct/assets/9853862/27d0674f-0e47-4c2c-9839-d9b9a2d03aa5

Note you get results relatively quickly that you can click on.

**Review by khalidadil — APPROVED** [review]

File: src/api/objects/ObjectAPI.js [hunk 1]
```diff
@@ -786,16 +786,17 @@ export default class ObjectAPI {
    * Given an identifier, constructs the original path by walking up its parents
    * @param {module:openmct.ObjectAPI~Identifier} identifier
    * @param {Array<module:openmct.DomainObject>} path an array of path objects
+   * @param {AbortSignal} abortSignal (optional) signal to abort fetch requests
    * @returns {Promise<Array<module:openmct.DomainObject>>} a promise containing an array of domain objects
    */
-  async getOriginalPath(identifier, path = []) {
-    const domainObject = await this.get(identifier);
+  async getOriginalPath(identifier, path = [], abortSignal = null) {
+    const domainObject = await this.get(identifier, abortSignal);
```
Comment (ozyx) [inline src/api/objects/ObjectAPI.js:793]: awesome

File: src/ui/layout/search/GrandSearch.vue [hunk 2]
```diff
@@ -130,45 +132,56 @@ export default {
...
         }
+      } finally {
+        if (this.abortSearchController) {
+          delete this.abortSearchController;
+        }
+      }
+    },
+    async searchObjects(abortSignal) {
+      const objectSearchPromises = this.openmct.objects.search(this.searchValue, abortSignal);
+      for await (const objectSearchResult of objectSearchPromises) {
```
Comment (ozyx) [inline src/ui/layout/search/GrandSearch.vue:161]: brilliant stuff!

**Review by ozyx — APPROVED** [review]
LGTM! A legit use of `for await`!

**Pull request merged**